### PR TITLE
pari: Make pthread optional, remove unstable

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

- make `pthread` optional because sage expects threading to be disabled and threading is disabled by default upstream
- remove unstable. Upstream apparently doesn't update unstable anymore and its already older than stable. Sage uses a git version of pari and I currently do that with an override, but I'm not sure whether or not that git version should be exposed outside of sage (@7c6f434c)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/math/pari/default.nix [hunk 1]
```diff
@@ -15,10 +17,10 @@ stdenv.mkDerivation rec {
```
Comment (matthewbauer) [inline pkgs/applications/science/math/pari/default.nix]: You can probably make libpthreadstubs optional as well

**Comment by 7c6f434c** [comment]
I don't use Pari/GP directly… intuitively maybe no, given that the Git revision is chosen based on Sage needs, not on some policy that only refers to Pari/GP development itself.

File: pkgs/applications/science/math/pari/default.nix (hunk 1 above)
Comment (timokau) [reply pkgs/applications/science/math/pari/default.nix]: You're right

**Comment by timokau** [comment]
Yeah I agree. Hopefully they'll switch back to stable releases anyways.

**Review by 7c6f434c — APPROVED** [review]

**Comment by 7c6f434c** [comment]
I hope I don't forget to come back and merge…

**Comment by matthewbauer** [comment]
Looks good! If you're not already, make sure you are checking Hydra jobs. This has much more extensive testing than done in PRs and occasionally we will pick up an unexpected failure:

https://hydra.nixos.org/jobset/nixos/trunk-combined

**Comment by timokau** [comment]
@matthewbauer I am to a certain extend (mostly only when hydra notifies me about stuff I maintain). But as long as we have hundreds of failing jobs, I don't think its a very useful indicator of which update broke something.

**Comment by matthewbauer** [comment]
Yeah definitely. I just know some new contributors are not aware of Hydra so I wanted to mention it. I mainly check the "newly failing jobs" tab.

**Pull request merged**